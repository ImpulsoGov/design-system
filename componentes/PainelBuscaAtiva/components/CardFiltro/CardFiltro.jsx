import React, { useState } from "react";
import style from "../../PainelBuscaAtiva.module.css";

export const CardFiltro = ({
  label,
  setOrdenar,
  ID,
})=>{
  const [property, setProperty] = useState("");
  const OrdenarPor = ()=>{
    // Atualiza estado local
    setProperty(ID[label]);
    // Atualiza estado do pai
    setOrdenar(ID[label]);
  }

  return (
    <div
      data-testid="CardFiltro"
      onClick={OrdenarPor}
      className={
        ID[label] !== property ?
        style.cardFiltro :
        style.cardFiltroSelected
      }
    >
      {label.toUpperCase()}
    </div>
  );
}