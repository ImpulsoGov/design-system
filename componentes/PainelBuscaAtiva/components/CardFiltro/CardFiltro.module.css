.cardFiltroSelected{
  font-family: 'Inter';
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 150%;
  background: #EFF5F9;
  width: fit-content;
  padding: 10px;
  border-radius: 10px;
  border: 1.5px solid #2EB380;
}

.cardFiltro{
  font-family: 'Inter';
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 150%;
  background: #EFF5F9;
  width: fit-content;
  padding: 10px;
  border-radius: 10px;
}

.cardFiltro:hover{
  border: 1.5px solid #2EB380;
  cursor: pointer;
}