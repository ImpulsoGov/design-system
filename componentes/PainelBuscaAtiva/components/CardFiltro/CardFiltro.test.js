
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { CardFiltro } from './CardFiltro';

const COMPONENT = 'CardFiltro';
const setOrdenar = jest.fn();
const scenarios = [
  {
    label: 'NOME DO ACS DE A-Z',
    setOrdenar,
    ordenar: '1',
    ID: { 'NOME DO ACS DE A-Z': 'acs_nome' }
  },
];

describe(`Componente: ${COMPONENT}`, () => {
  it('deve renderizar corretamente', () => {
    render(
      <CardFiltro
        {...scenarios[0]}
      />
    );
    const component = screen.getByTestId(COMPONENT);
    expect(component).toMatchSnapshot();
  });

  describe('Ao clicar no componente', () => {
    it('deve renderizar o card contornado', async () => {
      render(
        <CardFiltro
          {...scenarios[0]}
        />
      );
      const component = screen.getByTestId(COMPONENT);
      await userEvent.click(component);
      expect(screen.getByTestId(COMPONENT)).toMatchSnapshot();
    });

    it('deve chamar setOrdenar com o valor correto', async () => {
      render(
        <CardFiltro
          {...scenarios[0]}
        />
      );
      const component = screen.getByTestId(COMPONENT);
      await userEvent.click(component);
      expect(setOrdenar).toHaveBeenCalledWith('acs_nome');
    });
  });
});