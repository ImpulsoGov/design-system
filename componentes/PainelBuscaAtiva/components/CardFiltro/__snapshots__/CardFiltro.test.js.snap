// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente: CardFiltro Ao clicar no componente deve renderizar o card contornado 1`] = `
<div
  class="cardFiltroSelected"
  data-testid="CardFiltro"
>
  NOME DO ACS DE A-Z
</div>
`;

exports[`Componente: CardFiltro deve renderizar corretamente 1`] = `
<div
  class="cardFiltro"
  data-testid="CardFiltro"
>
  NOME DO ACS DE A-Z
</div>
`;
